suite-common/wallet-utils/src/transactionUtils.ts

```diff
@@ -1034,3 +1034,23 @@ export const groupTokensTransactionsByContractAddress = (
 
     return groupedTokensTxs;
 };
+
+export const getTransactionWithLowestNonce = (
+    transactionGroups: GroupedTransactionsByDate,
+): WalletAccountTransaction | null => {
+    let lowestNonceTransaction: WalletAccountTransaction | null = null;
+
+    Object.keys(transactionGroups).forEach(dateKey => {
+        transactionGroups[dateKey].forEach(transaction => {
```

Comment:
This code is fine, but it could be simplified :slightly_smiling_face: 

1. You can grab `Object.values` instead of `keys`
2. You can flatten the array, and avoid the nested `forEach`

```suggestion
    Object.values(transactionGroups)
        .flat()
        .forEach(transaction => {
```